### ISSUE TITLE
How to configure BERT TokenEmbedder so that it gets gradients?

### ISSUE THREAD

**Issue by aagohary** [allenai/allennlp#3756]
I see that ELMo tokenembedder has an argument called requires_grad. But that is not the case for BERT (PretrainedTransformerEmbedder).

**Comment by matt-gardner** [comment]
Are you saying that it's not getting updated?  What exactly are you trying to do that isn't working?

**Comment by aagohary** [comment]
I am trying to use allennlp-interpret and I get an empty "instance_1" output (no grad_input_1) .. I guess it has to do with the gradients at the embedding layer. I get proper outputs with a glove-based embedder as long as I set trainable to true.

**Comment by matt-gardner** [comment]
Can you give more details about what you're doing?  In particular, it'd be good if you could replicate what we do in the demo (which uses `PretrainedTransformerEmbedder`) in your setup, to help isolate the issue.  Also see this issue, which has some more info: https://github.com/allenai/allennlp/issues/3679.

**Comment by aagohary** [comment]
Thanks. I just added a `requires_grad: true` to the `token_embedders` config as 

```
token_embedders: {
        bert: {
          type: 'bert-pretrained',
          pretrained_model: 'bert-base-uncased',
          requires_grad: true
        }
      }
```
and that solved the problem with allennlp-interpret. The reason why I was confused was that I did not see `requires_grad` as a parameter in `PretrainedTransformerEmbedder`.

**Comment by matt-gardner** [comment]
I'm confused about why you need to pass that in; it seems to say that the default is `False`.  Oh!  You're using `bert-pretrained` instead of `pretrained_transformer` as your embedder type.  That's the problem.  Yeah, the default for `bert-pretrained` might have been `False`, but that's not true for `pretrained_transformer`.  The `bert-pretrained` one is also going away soon (it's already gone from master).

**Comment by aagohary** [comment]
I see. Thanks.
